solutions/cses-1144.mdx

```diff
@@ -19,64 +19,60 @@ TLE.
 using namespace std;
 using ll = long long;
 
-const int MX = 4e5 + 5;
+const int MAX = 4e5 + 5;
 
-ll bit[MX];
-vector<int> vals;
+ll bit[MAX]; // binary-indexed tree
+vector<int> vals; // values sorted for coordinate compression
 int n, q;
 
-void upd (int i, int val) {
-	for (; i <= MX; i += i&(-i)) {
-		bit[i] += val;
+void add(int value, int amount) {
+	int index = upper_bound(vals.begin(), vals.end(), value) - vals.begin();
+	for (; index <= MAX; index += index&(-index)) {
```

Comment:
this goes out of bounds. also, the parentheses around `(-index)` are not necessary